test/test_e2e.rb

```diff
@@ -104,4 +104,24 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert queue.fail_fast == 1
+    assert queue.unprocessed_jobs.length <= 2
```

Comment:
We should also test for `processed_jobs`:

```ruby
assert queue.processed_jobs.length <= 2
```

Reply:
Also, we should actually assert that the length is either 1 or 2. It shouldn't be 0 or negative.

Reply:
Length cant be negative AFAIK

Reply:
Since there are more tests in there it can actually reach up to 4